### ISSUE TITLE
aws-runas freezes for IAM user to assume a role

### ISSUE THREAD

**Issue by speedysingh786** [mmmorris1975/aws-runas#59]
migrated from 2.3.0 to 3.0.0
 **aws-runas non-prd aws s3 ls**  ask for MFA twice and freezes.
 
 After running debug **aws-runas -v  non-prd**  I see below log
 _DEBUG error caching credentials: rename C:\Users\XXXXXX\\.aws\.aws_session_token_default_363146724.tmp C:\Users\XXXXXX\\.aws\.aws_session_token_default: The process cannot access the file because it is being used by another process_.

I'm stuck at this, Any help ?

**Comment by mmmorris1975** [comment]
Thanks for raising this issue.  I don't think the fix will be very difficult, Windows manages files differently than Linux, which is what's causing this problem.

**Comment by mmmorris1975** [comment]
I saw similar messages on a Windows test server.  Made some updates, and the messages went away.  The credentials should now be cached as expected, and the extra MFA prompts should go away.  The new release is ready for you to try, let me know if this resolves the issue.

Fixed by: 12798c6d092dc7b104db373fc34754d304bde875
Released in [3.0.1](https://github.com/mmmorris1975/aws-runas/releases/tag/3.0.1)
